libexec/rtld-elf/aarch64/reloc.c

```diff
@@ -536,11 +536,24 @@ reloc_plt(Plt_Entry *plt, int flags, RtldLockState *lockstate)
 		case R_AARCH64_NONE:
 			break;
 #ifdef __CHERI_PURE_CAPABILITY__
+		/*
+		 * XXX Dapeng: The R_MORELLO_RELATIVE case should be removed
+		 * once the whole system is compiled with -cheri-codeptr-relocs,
+		 * for it would then be impossible to see this relocation here.
+		 */
 		case R_MORELLO_RELATIVE:
+		case R_MORELLO_FUNC_RELATIVE:
```

Comment:
Strictly speaking, shouldn't you add this new case in the earlier rtld commit and then refine it to add the comment above and the wrapping in this commit?  Actually, I wonder if the wrapping shouldn't in fact be added even before this commit (maybe I missed that when adding it, so maybe pull that out into a separate commit at the start of this PR to add the wrapping?)